pkg/metadata/gohai/gohai_test.go

```diff
@@ -34,3 +39,21 @@ func TestGetPayloadContainerized(t *testing.T) {
 	assert.Nil(t, gohai.Gohai.Network)
 	assert.NotNil(t, gohai.Gohai.Platform)
 }
+
+func TestGetPayloadContainerizedWithDocker0(t *testing.T) {
+	os.Setenv("DOCKER_DD_AGENT", "true")
+	defer os.Unsetenv("DOCKER_DD_AGENT")
+
+	detectDocker0()
```

Comment:
It is line needed?

Reply:
Yes, it's required to trigger the sync.Once so subsequent calls to `detectDocker0()` don't overwrite the global `docker0detected` value